utils/utils.go

```diff
@@ -210,3 +212,53 @@ func SortByPower(
 	}
 	return sortedOperatorAddrs, sortedPubKeys, sortedPowers
 }
+
+// AccumulateChanges accumulates the current and new validator updates and returns
+// a list of unique validator updates. The list is sorted by power in descending order.
+func AccumulateChanges(
+	currentChanges, newChanges []abci.ValidatorUpdate,
+) []abci.ValidatorUpdate {
+	// get only unique updates
+	m := make(map[string]abci.ValidatorUpdate)
+	for _, change := range currentChanges {
+		m[change.PubKey.String()] = change
+	}
+	for _, change := range newChanges {
+		m[change.PubKey.String()] = change
+	}
+
+	// convert to list
+	out := make([]abci.ValidatorUpdate, 0, len(m))
+	for _, update := range m {
+		out = append(out, update)
+	}
+
+	// The list of tendermint updates should hash the same across all consensus nodes
+	// that means it is necessary to sort for determinism.
+	sort.Slice(out, func(i, j int) bool {
+		if out[i].Power != out[j].Power {
+			return out[i].Power > out[j].Power
+		}
+		return out[i].PubKey.String() > out[j].PubKey.String()
+	})
+
+	return out
+}
+
+// AppendMany appends a variable number of byte slices together
+func AppendMany(byteses ...[]byte) (out []byte) {
```

Comment:
typo： bytes ？

Reply:
`byteses` as in multiple `bytes == []byte`. I will rename for clarity.